### PR TITLE
Fix `patch-package`

### LINKED ISSUE #6582: `patch-package` not correctly applied since Yarn upgrade
`patch-package` has not been applied to built binaries ever since #5555 was merged, leading to regressions coming from an unpatched `has-binary2` when users send certain circular data over WebSockets:

```
RangeError: Maximum call stack size exceeded
at get (internal/bootstrap/pre_execution.js:286:8)
at hasBinary (/builds/platform/ui/client/cache/Cypress/4.0.2/Cypress/resources/app/packages/socket/node_modules/has-binary2/index.js:44:3)
at hasBinary (/builds/platform/ui/client/cache/Cypress/4.0.2/Cypress/resources/app/packages/socket/node_modules/has-binary2/index.js:58:59)
at hasBinary (/builds/platform/ui/client/cache/Cypress/4.0.2/Cypress/resources/app/packages/socket/node_modules/has-binary2/index.js:58:59)
at hasBinary (/builds/platform/ui/client/cache/Cypress/4.0.2/Cypress/resources/app/packages/socket/node_modules/has-binary2/index.js:58:59)
```

The patch is correctly applied during development, but during production builds it is not. This is probably because `has-binary2` is in `node_modules` in dev, but in `packages/socket/node_modules` in the built binary.

### PR DESCRIPTION
**Pull request by flotwig**
- Closes #6582 

### User facing changelog

- Fixed a regression where Cypress could throw a "Maximum call stack size exceeded" error when transmitting a circular object over the internal websocket.

### Additional details

- don't hoist dependencies that are patched, this way we can be assured the path is always correct
- removed some dangerous `--ignore-engines` and `--silent`s
- [x] validated that the built packages contain the patch

### PR Tasks

- [na] Have tests been added/updated?
- [x] Has the original issue been tagged with a release in ZenHub? 
- [na] Has a PR for user-facing changes been opened in [`cypress-documentation`](https://github.com/cypress-io/cypress-documentation)? 
- [na] Have API changes been updated in the [`type definitions`](cli/types/index.d.ts)?
- [na] Have new configuration options been added to the [`cypress.schema.json`](cli/schema/cypress.schema.json)?

### CODE REVIEW

File: packages/driver/package.json [hunk 1]
```diff
@@ -66,6 +66,12 @@
...
   },
   "files": [
-    "lib"
-  ]
+    "lib",
+    "patches"
+  ],
+  "workspaces": {
+    "nohoist": [
+      "*"
```
Comment (andrew-codes) [inline packages/driver/package.json]: Does this mean that the root will now no longer hoist these driver packages while the building of the binary will keep them in the driver package? (Referencing https://github.com/cypress-io/cypress/pull/6583/files#diff-b9cfc7f2cdf78a7f4b91a753d10865a2L221-L223 in addition to this addition).

File: packages/socket/package.json [hunk 2]
```diff
@@ -5,23 +5,30 @@
   "main": "index.js",
   "scripts": {
     "clean-deps": "rm -rf node_modules",
+    "postinstall": "patch-package",
```
Comment (andrew-codes) [inline packages/socket/package.json]: Should `patch-package` now be moved from the root to the individual packages that use/require it? It was only added to the root to make it easier to work with, but it seems the opposite is actually true. What do you think about adding `patch-package` as a dev dependency to the packages that use it and remove it from the root `package.json`?

File: packages/socket/test/socket_spec.js [hunk 3]
```diff
@@ -18,7 +18,7 @@ describe('Socket', function () {
 
   context('.getPathToClientSource', function () {
     it('returns path to socket.io.js', function () {
-      const clientPath = path.join(resolvePkg('socket.io-client', { cwd: path.join(__dirname, '..', '..', '..') }), 'dist', 'socket.io.js')
+      const clientPath = path.join(resolvePkg('socket.io-client'), 'dist', 'socket.io.js')
```
Comment (andrew-codes) [inline packages/socket/test/socket_spec.js]: Nice. I like this.

File: packages/socket/package.json (hunk 2 above)
Comment (flotwig) [reply packages/socket/package.json]: I believe it needs to stay a root dependency - during binary-build, only production dependencies are installed, so patch-package wouldn't be installed at `postinstall`.

File: packages/driver/package.json (hunk 1 above)
Comment (flotwig) [reply packages/driver/package.json]: previously, `yarn install` from root and `yarn install` from packages/driver would "nohoist" these packages, but `yarn workspace driver install` would not. now, all 3 of those commands will have the same `nohoist` behavior.

Comment (flotwig) [reply packages/driver/package.json]: actually, i think the behavior is exactly the same as before. i made this change just in the interest of keeping package-specific config per-package as much as possible.

**Pull request merged**